src/card_number/mod.rs

```diff
@@ -1,42 +1,44 @@
-pub fn verify_card_number<T: AsRef<str>>(number: T) -> bool {
-    let number = number.as_ref();
-    let len = number.len();
-    if len < 16
-        || number.chars().any(|c| !c.is_numeric())
-        || number[1..10].parse::<u32>().unwrap() == 0
-        || number[10..16].parse::<u32>().unwrap() == 0
+trait CardVerification: AsRef<str> {
+    fn is_valid_card_number(&self) -> bool 
     {
-        return false;
-    }
+        let number = self.as_ref();
+        let len = number.len();
+        if len < 16
+            || number.chars().any(|c| !c.is_numeric())
+            || number[1..10].parse::<u32>().unwrap() == 0
+            || number[10..16].parse::<u32>().unwrap() == 0
+        {
+            return false;
+        }
 
-    let mut sum = 0;
-    let mut even;
-    let mut sub_digit;
-    for i in 0..16 {
-        even = if i % 2 == 0 { 2 } else { 1 };
-        sub_digit = number[i..i + 1].parse::<i32>().unwrap() * even;
-        sum += if sub_digit > 9 {
-            sub_digit - 9
-        } else {
-            sub_digit
-        };
+        let mut sum = 0;
+        let mut even;
+        let mut sub_digit;
+        for i in 0..16 {
+            even = if i % 2 == 0 { 2 } else { 1 };
+            sub_digit = number[i..i + 1].parse::<i32>().unwrap() * even;
+            sum += if sub_digit > 9 {
+                sub_digit - 9
+            } else {
+                sub_digit
+            };
+        }
+        sum % 10 == 0
     }
-    sum % 10 == 0
 }
 
+impl CardVerification for String {}
+impl CardVerification for &str {}
```

Comment:
I need to change it to `impl CardVerification for str {}` to support `CardVerification::is_valid_card_number("str")`. Currently, with `&str` I need to call it like `CardVerification::is_valid_card_number(&"str")`. @kianenigma Do you think it's fine to change it from `&str` to `str`?

Reply:
Not sure, I don't know what's the best practice here tbh 🤷 

but fyi, "str" is actually a @'static str, which means it is already a reference, so it looks strange that you need to reference it again as &"str". 

Looking a bit further, probably the issue is that the method is borrowing'&self', but changing that to 'self' won't play nicely with String either.

Reply:
For example, [to_string](https://doc.rust-lang.org/src/alloc/string.rs.html#2218-2223) is implemented for `str`. So I think it's a valid pattern.